src/app/util/ember-compatibility-functions.cpp

```diff
@@ -237,7 +238,39 @@ Protocols::InteractionModel::Status ServerClusterCommandExists(const ConcreteCom
         return Status::UnsupportedCluster;
     }
 
-    for (const CommandId * cmd = cluster->acceptedCommandList; cmd != nullptr; cmd++)
+    auto * commandHandler =
+        InteractionModelEngine::GetInstance()->FindCommandHandler(aCommandPath.mEndpointId, aCommandPath.mClusterId);
+    if (commandHandler)
+    {
+        struct Context
+        {
+            bool commandExists;
+            CommandId targetCommand;
+        } context{ false, aCommandPath.mCommandId };
+        CHIP_ERROR err = commandHandler->EnumerateAcceptedCommands(
+            aCommandPath,
+            [](CommandId command, void * context) -> Loop {
+                auto * ctx = static_cast<Context *>(context);
+                if (ctx->targetCommand == command)
+                {
+                    ctx->commandExists = true;
+                    return Loop::Break;
+                }
+                return Loop::Continue;
+            },
+            &context);
+        if (err != CHIP_ERROR_NOT_IMPLEMENTED)
+        {
+            if (err == CHIP_NO_ERROR)
+            {
+                return context.commandExists ? Status::Success : Status::UnsupportedCommand;
+            }
```

Comment:
This logic would benefit from a bit of general running commentary

Reply:
Added.